lighthouse-core/lib/large-javascript-libraries/bundlephobia-database.json

```diff
@@ -0,0 +1,158 @@
+{
+  "date-fns": {
+    "repository": "https://github.com/date-fns/date-fns.git",
+    "lastScraped": 1596239787705,
+    "versions": {
+      "2.15.0": {
+        "gzip": 18463
+      },
+      "latest": {
+        "gzip": 18463
+      },
+      "2.14.0": {
+        "gzip": 18304
+      },
+      "2.13.0": {
+        "gzip": 17928
+      },
+      "2.12.0": {
+        "gzip": 17484
+      },
+      "2.11.1": {
+        "gzip": 17450
+      },
+      "2.11.0": {
+        "gzip": 17406
+      },
+      "2.10.0": {
+        "gzip": 17385
+      },
+      "2.9.0": {
+        "gzip": 17355
+      },
+      "2.8.1": {
+        "gzip": 17166
+      },
+      "2.8.0": {
+        "gzip": 17165
+      }
+    }
+  },
+  "luxon": {
+    "repository": "https://github.com/moment/luxon.git",
+    "lastScraped": 1596239802128,
+    "versions": {
+      "1.24.1": {
```

Comment:
sorry sorry I missed that

1. thx, fixed

![image](https://user-images.githubusercontent.com/66381097/89681336-621ff680-d8ba-11ea-8b89-a33fdca984c8.png)

2. why?

Reply:
> 2. why?

We're not even sure these sizes are accurate for the way that they are serving and compressing the library. They might be tree-shaking and be a fraction of this size or they might not be compressing at all and the size is really 6x larger than this. The sizes are *very rough approximations* and if it's a very rough approximation anyway, then increasing the lighthouse bundle size for a savings difference of a few bytes isn't space well-spent.

It's less of a big deal now that you've addressed the no extra versions of the other libraries issue though.

Reply:
That's true but, all else equal, the closest approximation that we have for the _true_ size of a particular version of a library _is_ what BundlePhobia reports. If we're defaulting to `latest` anyway, that would be equally error-prone, assuming that BundlePhobia's numbers are inaccurate. If they _are_ accurate, then we're throwing away data precision to save a few kb which feels like a tradeoff we shouldn't make until we _prove_ that BP isn't accounting for something like compressing.

Especially since we don't know if it's an overestimate or underestimate, all we can be certain of is that the _relative_ size difference between particular versions of a library is accurate, since the same statistical error would affect each library version the same way (for the most part), so the highest confidence we can have in the data is actually the difference between each library version, even if that difference is small. That's why I think we should store each library version separately. If, at some point, the size of the file becomes too large, we could revisit this conversation as well.

Thoughts?

Reply:
> If we're defaulting to latest anyway, that would be equally error-prone, assuming that BundlePhobia's numbers are inaccurate.

That's kind of exactly my point, these sizes are all so hand-wavvy anyway. Why are we trying to be accurate to the byte with library version information when the user will

> we're throwing away data precision 

Data precision that the end user of Lighthouse never sees ;)

> we prove that BP isn't accounting for something like compressing.

I'm not sure I understand this. We already have proof. They're reporting the gzipped size and the websites exist that do not serve gzipped content (could be brotli or uncompressed).

Reply:
> That's why I think we should store each library version separately

To be clear, we absolutely should. I'm just saying if the gzipped size of a specific version of the library is within a few bytes of the *fallback* then don't both saving that specific version information.

Reply:
Ohh yeah that totally makes sense - I forgot about the rounding. Let me add the change :)